openhands/sdk/conversation/conversation_stats.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional
+
+from pydantic import BaseModel, Field, PrivateAttr
+
+from openhands.sdk.conversation.types import ConversationID
+from openhands.sdk.io.base import FileStore
+from openhands.sdk.llm.llm_registry import RegistryEvent
+from openhands.sdk.llm.utils.metrics import Metrics
+from openhands.sdk.logger import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class ConversationStats(BaseModel):
+    # Public fields that will be serialized
+    service_to_metrics: dict[str, Metrics] = Field(
+        default_factory=dict,
+        description="Active service metrics tracked by the registry",
+    )
+
+    _restored_services: set = PrivateAttr(default_factory=set)
+
+    # Private attributes that won't be serialized
+    _file_store: Optional[FileStore] = PrivateAttr(default=None)
+    _conversation_id: Optional[ConversationID] = PrivateAttr(default=None)
+    _metrics_file_name: str = PrivateAttr(default="conversation_stats.pkl")
+
+    def __init__(
+        self,
+        file_store: FileStore | None = None,
+        conversation_id: ConversationID | None = None,
+    ):
+        super().__init__()
```

Comment:
@openhands lets not create an `__init__` method for this class. Instead create a classmethod with the name `create`. This should take the relevant params, set the private attributes, call `maybe_restore_metrics` and return the conversationstats class 

remove the init method completely 

don't write or rewrite any unit tests for now, we'll look at them later

if you want to lint the files, make sure to remove openhands/code from your pythonpath first